src/answers-search-bar.js

```diff
@@ -0,0 +1,519 @@
+   * so we do a regular set instead of a setWithPersist here.
+   *
+   * @param {SearchConfig} searchConfig
+   * @private
+   */
+  _setDefaultInitialSearch (searchConfig) {
+    if (searchConfig.defaultInitialSearch == null) {
+      return;
+    }
+    const prepopulatedQuery = this.core.storage.get(StorageKeys.QUERY);
+    if (prepopulatedQuery != null) {
+      return;
+    }
+    this.core.storage.set(StorageKeys.QUERY_TRIGGER, QueryTriggers.INITIALIZE);
+    this.core.storage.set(StorageKeys.QUERY, searchConfig.defaultInitialSearch);
+  }
+
+  /**
+   * Sets the geolocation tag in storage, overriding other inputs. Do not use in conjunction
+   * with other components that will set the geolocation internally.
+   * @param {number} lat
+   * @param {number} long
+   */
+  setGeolocation (lat, lng) {
+    this.core.storage.set(StorageKeys.GEOLOCATION, {
+      lat, lng, radius: 0
+    });
+  }
+
+  /*
+   * Adds context as a parameter for the query API calls.
+   * @param {Object} context The context object passed in the API calls
+   */
+  setContext (context) {
+    const contextString = JSON.stringify(context);
+    if (!isValidContext(contextString)) {
+      console.error(`Context parameter "${context}" is invalid, omitting from the search.`);
+      return;
+    }
+
+    this.core.storage.set(StorageKeys.API_CONTEXT, contextString);
+  }
+
+  /**
+   * Gets the locale that ANSWERS was initialized to
+   *
+   * @returns {string}
+   */
+  _getInitLocale () {
+    return this.core.storage.get(StorageKeys.LOCALE);
+  }
+
+  /**
+   * Parses a value from persistent storage, which stores strings,
+   * into the shape the SDK expects.
+   * TODO(SLAP-1111): Move this into a dedicated file/class.
+   *
+   * @param {string} key
+   * @param {string} value
+   * @returns {string|number|Filter}
+   */
+  _parsePersistentStorageValue (key, value) {
+    switch (key) {
+      case StorageKeys.PERSISTED_FILTER:
+        return Filter.from(JSON.parse(value));
+      case StorageKeys.PERSISTED_LOCATION_RADIUS:
+        return parseFloat(value);
+      case StorageKeys.PERSISTED_FACETS:
+      case StorageKeys.SORT_BYS:
+        return JSON.parse(value);
+      default:
+        return value;
+    }
+  }
+}
+
+/**
+ * @param {Object} config
+ * @param {Storage} storage
+ * @returns {Services}
+ */
+function getServices (config, storage) {
+  return {
+    errorReporterService: new ErrorReporter(
+      {
+        apiKey: config.apiKey,
+        experienceKey: config.experienceKey,
+        experienceVersion: config.experienceVersion,
+        printVerbose: config.debug,
+        sendToServer: !config.suppressErrorReports,
+        environment: config.environment
+      },
+      storage)
+  };
+}
+
+/**
+ * Initialize the scroll event listener to send analytics events
+ * when the user scrolls to the bottom. Debounces scroll events so
+ * they are processed after the user stops scrolling
+ */
+function initScrollListener (reporter) {
```

Comment:
From looking at the code, I thought this would be fired on the client's main page (which seems like incorrect behavior). But then I tried this on a few live client pages and it doesn't look like that's the case. Is there any way we could get rid of this? scroll listeners generally aren't great performance wise